backend/de.metas.manufacturing/src/main/java/org/eevolution/api/impl/PPOrderRoutingRepository.java

```diff
@@ -103,11 +110,26 @@ public PPOrderRouting getByOrderId(@NonNull final PPOrderId orderId)
 				//
 				.firstActivityCode(firstActivityCode)
 				.activities(orderActivities)
+				.products(orderProducts)
 				.codeToNextCodeMap(codeToNextCodeMap)
 				//
 				.build();
 	}
 
+	private PPOrderRoutingProduct toPPOrderRoutingProduct(final I_PP_Order_Node_Product product)
+	{
+		final PPOrderId orderId = PPOrderId.ofRepoId(product.getPP_Order_ID());
+		final PPOrderRoutingActivityId activityId = PPOrderRoutingActivityId.ofRepoId(orderId, product.getPP_Order_Node_ID());
+		final PPOrderRoutingProductId id = PPOrderRoutingProductId.ofRepoId(activityId, product.getPP_Order_Node_Product_ID());
+		return PPOrderRoutingProduct.builder()
+				.id(id)
+				.productId(ProductId.ofRepoId(product.getM_Product_ID()))
+				.qty(product.getQty())
```

Comment:
i suggest to use `Quantity` as type for `qty` instead of `BigDecimal`.
You can assume the qty's UOM is the stocking uom.
You can get it using `IProductBL.getStockUOM`.